src/ui/pages/common/PageTreeActionMenu.js

```diff
@@ -24,10 +24,17 @@ class PageTreeActionMenu extends Component {
       onClickClone, onClickDelete, onClickPublish, onClickUnpublish,
     } = this.props;
 
+    let disabled = false;
+    if (!page.isEmpty && page.status === PAGE_STATUS_PUBLISHED) {
+      disabled = true;
+    }
+    if (page.expanded) {
+      disabled = page.pagePublish;
+    }
     const changePublishStatus = page.status === PAGE_STATUS_PUBLISHED ?
       (
         <MenuItem
-          disabled={page.status === PAGE_STATUS_PUBLISHED && !page.isEmpty}
+          disabled={disabled}
```

Comment:
This is not the change that was required in the ticket IMHO.

> Disable "unpublish" when the page has even only one public child page
> Disable "delete" whatever its status, when a page has children pages

So, I think you should : 

1. in the UNPUBLISH button: ` disabled={page.hasPublishedChildren} `
1. in the DELETE button: ` disabled={!page.isEmpty} `